appserver/distributions/glassfish-common/src/main/resources/bin/asadmin.bat

```diff
@@ -15,20 +16,21 @@ REM
 REM  SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
 REM
 
-
-REM  Always use JDK 1.6 or higher
-REM  Depends on Java from ..\glassfish\config\asenv.bat
 VERIFY OTHER 2>nul
 setlocal ENABLEEXTENSIONS
 if ERRORLEVEL 0 goto ok
 echo "Unable to enable extensions"
 exit /B 1
+
 :ok
-call "%~dp0..\glassfish\config\asenv.bat"
+set AS_INSTALL=%~dp0..\glassfish
+call "%AS_INSTALL%\config\asenv.bat"
 if "%AS_JAVA%x" == "x" goto UsePath
-set JAVA="%AS_JAVA%\bin\java"
+set JAVA=%AS_JAVA%\bin\java

```

Comment:
Quotes really not needed here?

Reply:
Ok, I learned how it is done in BAT file - you wrap whole argument of the SET command. So the original was wrong, this was better, but safest option is to wrap it all.